custom/icds_reports/utils/data_accessor.py

```diff
@@ -48,3 +49,47 @@ def get_program_summary_data_with_retrying(step, domain, config, now, include_te
             sleep(5)
             retry += 1
     return data
+
+
+# keeping cache timeout as 2 hours as this is going to be used
+# in some script/tool which might flood us with requests
+@icds_quickcache(['state_id', 'month'], timeout=120 * 60)
+def get_disha_api_v2_data(state_id, month):
+
+    data = DishaIndicatorViewV2.objects.filter(month=month,
+                                               state_id=state_id).first()
+
+    #not using lxml because using lxml when no xml manipulation needed is just overengineering
+    api_xml_response = """
+    <?xml version="1.0" encoding="UTF-8"?>
+    <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://www.w3.org/2001/12/soap-envelope"
+    SOAP-ENV:encodingStyle="http://www.w3.org/2001/12/soap-encoding">
+       <SOAP-ENV:Header />
+       <SOAP-ENV:Body>
+          <state>{state_name}</state>
+          <month>{month}</month>
+          <num_launched_awcs>{num_launched_awcs}</num_launched_awcs>
+          <num_households_registered>{cases_household}</num_households_registered>
+          <pregnant_enrolled>{cases_ccs_pregnant}</pregnant_enrolled>
+          <lactating_enrolled>{cases_ccs_lactating}</lactating_enrolled>
+          <children_enrolled>{cases_child_health}</children_enrolled>
+          <bf_at_birth>{bf_at_birth}</bf_at_birth>
+          <ebf_in_month>{ebf_in_month}</ebf_in_month>
+          <cf_in_month>{cf_in_month}</cf_in_month>
+       </SOAP-ENV:Body>
+    </SOAP-ENV:Envelope>
+    """.format(**{
```

Comment:
I think this is slightly clearer just using named arguments i.e. (`format(state_name=data.state_name, month=data.month)` etc)

Reply:
Oww ok..